packages/amplify-appsync-simulator/src/velocity/util/errors.ts

```diff
@@ -1,24 +1,47 @@
+import { GraphQLResolveInfo } from 'graphql';
+
 export class TemplateSentError extends Error {
-  constructor(gqlMessage, errorType, data, errorInfo) {
-    super(gqlMessage);
-    Object.assign(this, { gqlMessage, errorType, data, errorInfo });
+  extensions: any;
+  constructor(message, errorType, data, errorInfo, info: GraphQLResolveInfo) {
+    super(message);
+    const fieldName = info.fieldName;
+    let path = info.path;
+    const pathArray = []
+    do {
+      pathArray.splice(0, 0, path.key);
+      path = path.prev
+    } while(path);
+
+    const fieldNode = info.fieldNodes.find(f => f.name.value === fieldName);
+    const filedLocation = fieldNode && fieldNode.loc.startToken || null;
+    Object.assign(this, { message, errorType, data, errorInfo });
```

Comment:
Instead of this can't we could use [Typescript constructor assignment](https://kendaleiv.com/typescript-constructor-assignment-public-and-private-keywords/)?